### ISSUE TITLE
get_flash_videos loops forever eating all processor ressources

### ISSUE THREAD

**Issue by GoogleCodeExporter** [MiraculousLegendas/get-flash-videos#114]
```
What steps will reproduce the problem?
get_flash_videos --debug
'http://vids.myspace.com/index.cfm?fuseaction=vids.individual&videoid=103893882'

What is the expected output? What do you see instead?
Exit without looping forever. The get_flash_videos perl process eats up as
much processor time it can get.

What version of the product are you using? On what operating system?
get_flash_videos version 1.21 (http://code.google.com/p/get-flash-videos/).
opensuse 11.0 and 11.1

Please provide any additional information below.
When executing the command line given above, I see a lot of output from
get_flash_video trying to fetch that video. The process then hangs without
exiting at the following lines:
Using method 'generic' for
http://vids.myspace.com/index.cfm?fuseaction=vids.channel&vanity=Ir
-> GET http://vids.myspace.com/index.cfm?fuseaction=vids.channel&vanity=Ir
<- 206 text/html; charset=utf-8 (9972): UTF8 on, non-ASCII, 9972 characters
10001 bytes

I have to kill the process manually. This error is not limited to myvideo I
also noted it on other websites that get_flash_videos does not support. I
don't have a problem with not supporting the video site, the problem is
with looping forever and eating my processor performance ;-)
```

Original issue reported on code.google.com by `ra...@dubistmeinheld.de` on 18 Apr 2010 at 6:52

**Comment by GoogleCodeExporter** [comment]
```
Looks like a WWW::Mechanize bug, haven't been able to work it out yet.
```

Original comment by `monsieur...@gmail.com` on 9 May 2010 at 1:30
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
I also found out that in the combined version of get_flash_videos 1.20, the 
program
exits properly.
```

Original comment by `ra...@dubistmeinheld.de` on 9 May 2010 at 5:39

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Hi,

any progress on this one? ;-)

I have tried version 1.23 and we have still the same problem. I'm stuck with 
1.20 on my server.
```

Original comment by `ra...@dubistmeinheld.de` on 6 Aug 2010 at 8:00

**Comment by GoogleCodeExporter** [comment]
```
I have seen a hang like this on a few sites when using get_flash_videos 1.23.  
Using the perl debugger, I traced it down to this else block in 
FlashVideo::Generic::find_video:

    # Extract URLs in a frail way.
    while ($browser->content =~ m{(http://[^ "']+?\.swf)}ig) {
      $swf_urls{$_[1]}++;
    }

Beyond the issue that this block never finishes, there is a secondary problem: 
the choice of hash key is incorrect.  The value of $_[1] is a hash reference 
passed by the caller.  Since the code subsequently does "foreach my $swf_url 
(keys %swf_urls)" and uses $swf_url in a context that looks string oriented, I 
think the proper hash key in the loop is $1, the value matched and captured 
from m{}.  The use of $_[1] is probably a copy&paste artifact from the 
anonymous subroutine used when URI::Find is available.

As an additional factor that can complicate debugging of this problem, once I 
installed URI::Find, get_flash_videos switched to using the associated if() 
branch, which does _not_ hang when presented with the same page that broke the 
else branch.  Thus, anyone wishing to pursue this problem must not have 
URI::Find available.  (Even with URI::Find, I was unable to get the video, but 
get_flash_videos did exit gracefully with an error that it was unable to find 
any videos.)
```

Original comment by `google.8...@spamgourmet.com` on 9 Oct 2010 at 3:43

**Comment by GoogleCodeExporter** [comment]
```
Thanks a lot for comment #5, this indeed is the problem. Installing URI::Find 
fixes the issue.

But unfortunately this dependency is unknown by the manual and most users don't 
know about this issue when they haven't install URI::Find beforehand. Maybe the 
best way is to remove the "# Extract URLs in a frail way" block in the code and 
require URI::Find.
```

Original comment by `ra...@dubistmeinheld.de` on 2 Dec 2010 at 5:12

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the investigation, that's very helpful :)

I'll fix the broken code; we aim to not require too many modules (although I'll 
recommend it somewhere).
```

Original comment by `monsieur...@gmail.com` on 3 Dec 2010 at 5:53

**Comment by GoogleCodeExporter** [comment]
```
Fixed by git 7f56b5d.
```

Original comment by `monsieur...@gmail.com` on 3 Dec 2010 at 6:09
- Changed state: **Fixed**
